### PR TITLE
Polling sub tasks

### PR DESCRIPTION
**Pull request by adamruzicka**
- [X] resuming
- [x] tests
  - [X] tests for polling
  - [x] tests for polling + bulk
  - [x] tests for resume

### CODE REVIEW

File: lib/dynflow/action/with_polling_sub_plans.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    end
+
+    def poll
+      recalculate_counts
+      try_to_finish || suspend_and_ping
+    end
+
+    def wait_for_sub_plans(sub_plans)
+      increase_counts(sub_plans.count, 0)
+      if is_a?(::Dynflow::Action::WithBulkSubPlans)
```
Comment (iNecas) [inline lib/dynflow/action/with_polling_sub_plans.rb]: Why do we need this check?

Comment (adamruzicka) [inline lib/dynflow/action/with_polling_sub_plans.rb]: We don't really need it, but it makes sense to start polling after all the sub plans were spawned.

File: lib/dynflow/action/with_polling_sub_plans.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      suspend { |suspended_action| ping suspended_action }
+    end
+
+    def ping(suspended_action)
+      world.clock.ping suspended_action, REFRESH_INTERVAL, Poll
+    end
+
+    def recalculate_counts
+      total      = sub_plans.count
+      @sub_plans = nil # TODO:
```
Comment (iNecas) [inline lib/dynflow/action/with_polling_sub_plans.rb]: Planning to clean this TODOs, right? :)

File: test/action_test.rb [hunk 3]
```diff
@@ -442,6 +481,34 @@ def run(event = nil)
           resumed_plan.state.must_equal :stopped
           resumed_plan.result.must_equal :success
         end
+
+        describe ::Dynflow::Action::WithPollingSubPlans do
+          include TestHelpers
+
+          let(:klok) { Dynflow::Testing::ManagedClock.new }
```
Comment (iNecas) [inline test/action_test.rb]: Any reason not to call it `:clock`?

**Comment by iNecas** [comment]
Codewise looks good, waiting for the test of the tests to start testing this manually

File: test/action_test.rb (hunk 3 above)
Comment (adamruzicka) [inline test/action_test.rb]: not really, will change it

File: lib/dynflow/action/with_polling_sub_plans.rb (hunk 2 above)
Comment (iNecas) [inline lib/dynflow/action/with_polling_sub_plans.rb]: If they are dependent, having two commits in one PR, with the final set of changes would be the best.

Comment (adamruzicka) [inline lib/dynflow/action/with_polling_sub_plans.rb]: Both PRs should now containing commit https://github.com/Dynflow/dynflow/pull/243/commits/fdf57ed3273f141a4bf56687072f1b3eddeb6ca5 with the common changes

**Comment by iNecas** [comment]
Tested and works well. Thanks @adamruzicka. Please crate an issue for foreman_remote_execution to start using it and let's add it there. I think it's a prerequisity for http://projects.theforeman.org/issues/19937/ to work as expected (from the user point of view)

**Pull request merged**